src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -198,14 +208,17 @@ private async Task SynchronizeSuppressedIssuesAsync()
                 cancellationTokenSource?.Cancel();
                 cancellationTokenSource = new CancellationTokenSource();
 
+                var serverBranch = await serverBranchProvider.GetServerBranchNameAsync();
```

Comment:
We should really have that rule that async methods need to receive cancellation tokens... 
There is cancellationTokenSource.Token right below that we could use. Does it make sense to pass it to GetServerBranchNameAsync?

Reply:
Yes, probably. I'll add it later.